src/bootstrap/flags.rs

```diff
@@ -239,25 +232,32 @@ Arguments:
             }
         });
 
-        // All subcommands can have an optional "Available paths" section
-        if matches.opt_present("verbose") {
-            let flags = Flags::parse(&["build".to_string()]);
-            let mut config = Config::parse(&flags.build, cfg_file.clone());
-            config.build = flags.build.clone();
-            let mut build = Build::new(flags, config);
-            metadata::build(&mut build);
-            let maybe_rules_help = step::build_rules(&build).get_help(subcommand);
-            if maybe_rules_help.is_some() {
-                extra_help.push_str(maybe_rules_help.unwrap().as_str());
-            }
-        } else {
-            extra_help.push_str(format!("Run `./x.py {} -h -v` to see a list of available paths.",
-                     subcommand).as_str());
-        }
+        // FIXME: The rustbuild rewrite does not make an available paths
+        // sections simple or easy to implement. I'm not sure how useful it is
+        // to have so perhaps we shouldn't attempt to reimplement it due to the
+        // difficulties imposed by such a move.
```

Comment:
Yeah, I thought about the vector or maybe iterator of paths idea. I wasn't aware that this was actually used by anyone, if it is we should definitely implement it. I'm going to go with the vector of paths idea for now since the ShouldRun struct seems possibly more complicated, but I'll assess both options before finalizing which one I choose to implement.